pix2tex/dataset/latex2png.py

```diff
@@ -122,9 +140,32 @@ def tex2png(eq, **kwargs):
 
 
 def tex2pil(tex, **kwargs):
-    pngs = Latex(tex, **kwargs).write(return_bytes=True)
+    pngs, error_index = Latex(tex, **kwargs).write(return_bytes=True)
     images = [Image.open(io.BytesIO(d)) for d in pngs]
-    return images
+    return images, error_index
+
+
+def extract(text, expression=None, type: str = None):
+    """extract text from text by regular expression
+
+    Args:
+        text (str): input text
+        expression (str, optional): regular expression. Defaults to None.
+        type (str, optional): type of extracted text. Defaults to None.
+
+    Returns:
+        str: extracted text
+    """
+    if type is not None:
+        type2expression = {"en": r"[a-zA-Z]+", "zh": r"[\u4e00-\u9fa5]+", "num": r"\d+",
+                           "punctuation": u"[\u3002\uff1b\uff0c\uff1a\u201c\u201d\uff08\uff09\u3001\uff1f\u300a\u300b]"}
+        expression = type2expression[type]
```

Comment:
What is that part for? it is never used

Reply:
yeah, that's part useless.

---

thanks for your review, another reason I want to make contributions is hope someone like you can review my code, and point out deficiencies that I am not yet aware of. 🤣 

You are shortage at time and I am shortage at the chance. if you want to do something but haven't enough time, I think I can do something for you (deep learning area).  I want to get in touch with good people like you. I need some chance to grow.